components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/AuthReqBuilder.java

```diff
@@ -106,4 +185,150 @@ private static NameIDPolicy buildNameIDPolicy() {
         nameIDPolicy.setAllowCreate(true);
         return nameIDPolicy;
     }
+
+    /**
+     * Sign the SAML AuthnRequest message
+     *
+     * @param authnRequest
+     * @param signatureAlgorithm
+     * @param cred
+     * @return
+     */
+    public static AuthnRequest setSignature(AuthnRequest authnRequest, String signatureAlgorithm,
+            X509Credential cred) throws Exception {
```

Comment:
Fixed throwing generic exception in #849, #1523 (https://github.com/wso2/carbon-apimgt/commit/a3c032ab9f7593c6e11cdf1e91fa04d5cc3de991)